docs/usage/MTCAPIDoc.md

```diff
@@ -0,0 +1,417 @@
+# MTC API Documentation
+
+MTC has several APIs which are needed to carry out migrations. This document explains specs for each API used in migrations.
+
+## Direct Image Migration
+
+```
+apiVersion: migration.openshift.io/v1alpha1
+kind: DirectImageMigration
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: directimagemigration-sample
+spec:
+  srcMigClusterRef:
+    name: migcluster-local
+    namespace: openshift-migration
+
+  destMigClusterRef:
+    name: migcluster-remote
+    namespace: openshift-migration
+
+  namespaces:
+  - nginx-example
+```
+
+-   spec.destMigClusterRef: ObjectReference contains enough information to let you inspect or modify the referred object. Contains name and namespace of the destination mig cluster where the operator is installed.
+
+-   spec.namespaces:  A list of all namespaces to run DIM to get all the imagestreams in these namespaces.
+
+-   spec.srcMigClusterRef: ObjectReference contains enough information to let you inspect or modify the referred object. Contains name and namespace of the source mig cluster where the operator is installed.
+
+## Direct ImageStream Migration
+
+```
+apiVersion: migration.openshift.io/v1alpha1
+kind: DirectImageStreamMigration
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: directimagestreammigration-sample
+spec:
+  srcMigClusterRef:
+    name: migcluster-local
+    namespace: openshift-migration
+
+  destMigClusterRef:
+    name: migcluster-remote
+    namespace: openshift-migration
+
+  imageStreamRef:
+    name: nginx
+    namespace: nginx-example
+
+  destNamespace: new-namespace
+  ```
+
+- spec.destMigClusterRef: ObjectReference contains enough information to let you inspect or modify the referred object. Contains name and namespace of the destination mig cluster where the operator is installed.
+
+- spec.destNamespace: A string holding the name of the namespace on destination cluster where imagestreams should be migrated.
+
+- spec.imageStreamRef: ObjectReference contains enough information to let you inspect or modify the referred object. Contains name and namespace of the source mig cluster where the imagesstream is present.
```

Comment:
@JaydipGabani 
> spec.imageStreamRef: .... Contains name and namespace of the source mig cluster where the imagesstream is present.

This does not look right. It sounds like "name" is the source cluster name. However, the example looks like "name" refers to the image stream name. Can you confirm?